src/vs/platform/files/common/files.ts

```diff
@@ -555,6 +555,14 @@ export interface IFilesConfiguration {
 		eol: string;
 		hotExit: string;
 		useExperimentalFileWatcher: boolean;
+		nesting: {
+			enable: boolean;
+			rules: {
+				[glob: string]: boolean | {
+					when: string | string[];
+				};
+			};
+		};
```

Comment:
That is limited to nesting by design. It means that it will be inconsistent to add new virtual elements (think of virtual directory that does not exist)